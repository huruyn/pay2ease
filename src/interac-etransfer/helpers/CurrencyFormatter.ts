const currencyFormatter = (currency = 'USD') => (
  new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency,
  }));

export default currencyFormatter;
