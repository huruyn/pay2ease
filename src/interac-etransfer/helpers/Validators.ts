import { PhoneNumberUtil } from 'google-libphonenumber';

const emailRegExp = /^[a-zA-Z0-9.!#$%&â€™*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;

export const isValidEmail = (data: string): boolean => emailRegExp.test(data);

export const isValidPhone = (data: string): boolean => {
  const phoneUtil = PhoneNumberUtil.getInstance();
  try {
    const number = phoneUtil.parse(data);
    return phoneUtil.isValidNumber(number);
  } catch (e) {
    return false;
  }
};
