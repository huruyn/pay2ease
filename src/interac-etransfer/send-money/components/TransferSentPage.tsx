import { DateTime } from 'luxon';
import {
  Row, Col, Form, Button,
} from 'react-bootstrap';
import { useNavigate } from 'react-router-dom';
import { PageIds } from '../SendMoney';
import currencyFormatter from '../../helpers/CurrencyFormatter';

export interface TransferInformation{
  source: { name: string; email: string };
  destination: { name: string; email: string };
  fromAccount: string;
  securityQuestion: string;
  securityAnswer: string;
  expiryDate: DateTime;
  submitted: DateTime;
  balance: number;
  amount: number;
  referenceNumber: number;
}
interface TransferSentPageProps{
  isCompleted: boolean;
  transferInformation: TransferInformation;
}

const TransferSentPage = ({
  isCompleted,
  transferInformation,
}: TransferSentPageProps): JSX.Element => {
  const navigate = useNavigate();

  return (
    <>
      <div className="d-flex align-items-center">
        <div
          style={{
            width: 20,
            height: 25,
            border: '1px dotted grey',
            textAlign: 'center',
            marginRight: 10,
          }}
        >
          P
        </div>
        <h4>Your transfer has been sent</h4>
      </div>
      <p>
        If you opted to send a notification of your transfer, your contact will
        receive it shortly.
      </p>
      {!isCompleted && <p>This transfer is final and can&apos;t be recalled.</p>}
      <Row className="mt-4">
        <Col md={4}>
          <Form.Label>From:</Form.Label>
        </Col>
        <Col md={8}>
          <Form.Label>{transferInformation.source.name}</Form.Label>
          <p>{transferInformation.source.email}</p>
        </Col>
      </Row>
      <hr style={{ height: '1px', width: '100%' }} />
      <Row className="mt-2">
        <Col md={4}>
          <Form.Label>To:</Form.Label>
        </Col>
        <Col md={8}>
          <Form.Label>{transferInformation.destination.name}</Form.Label>
          <p>{transferInformation.destination.email}</p>
        </Col>
      </Row>
      <hr style={{ height: '1px', width: '100%' }} />
      <Row className="mt-2">
        <Col md={4}>
          <Form.Label>Account Balance:</Form.Label>
        </Col>
        <Col md={8}>
          <Form.Label>{`${currencyFormatter().format(transferInformation.balance)}`}</Form.Label>
        </Col>
      </Row>
      <hr style={{ height: '1px', width: '100%' }} />
      {isCompleted && (
        <>
          <Row className="mt-2">
            <Col md={4}>
              <Form.Label>Security Question:</Form.Label>
            </Col>
            <Col md={8}>
              <Form.Label>{transferInformation.securityQuestion}</Form.Label>
            </Col>
          </Row>
          <hr style={{ height: '1px', width: '100%' }} />
          <Row className="mt-2">
            <Col md={4}>
              <Form.Label>Response:</Form.Label>
            </Col>
            <Col md={8}>
              <Form.Label>***************</Form.Label>
            </Col>
          </Row>
          <hr style={{ height: '1px', width: '100%' }} />
        </>
      )}

      <Row className="mt-2">
        <Col md={4}>
          <Form.Label>Amount:</Form.Label>
        </Col>
        <Col md={8}>
          <Form.Label>{`${currencyFormatter().format(transferInformation.amount)}`}</Form.Label>
        </Col>
      </Row>
      <hr style={{ height: '1px', width: '100%' }} />
      <Row className="mt-2">
        <Col md={4}>
          <Form.Label>From Account:</Form.Label>
        </Col>
        <Col md={8}>
          <Form.Label>{transferInformation.fromAccount}</Form.Label>
        </Col>
      </Row>
      <hr style={{ height: '1px', width: '100%' }} />
      {isCompleted && (
        <>
          <Row className="mt-2">
            <Col md={4}>
              <Form.Label>Expiry Date:</Form.Label>
            </Col>
            <Col md={8}>
              <Form.Label>
                {transferInformation.expiryDate.toLocaleString(DateTime.DATE_MED)}
              </Form.Label>
            </Col>
          </Row>
          <hr style={{ height: '1px', width: '100%' }} />
        </>
      )}
      <Row className="mt-2">
        <Col md={4}>
          <Form.Label>Reference Number:(keep for your records)</Form.Label>
        </Col>
        <Col md={8}>
          <Form.Label>{transferInformation.referenceNumber}</Form.Label>
        </Col>
      </Row>
      <p className="mt-4">{`Submitted: ${transferInformation.submitted.toLocaleString(DateTime.DATETIME_MED_WITH_WEEKDAY)}.`}</p>
      <hr style={{ height: '1px', width: '100%' }} />
      <Row className="mt-4 mb-4">
        <Col className="d-flex justify-content-end">
          <Button
            variant="outline-danger"
            className="d-flex"
            style={{ marginRight: 10 }}
            onClick={() => {
              navigate('/interac-etransfer/status/sent');
            }}
          >
            Check Status
          </Button>
          <Button
            className="zippy-btn d-flex"
            onClick={() => {
              navigate(`/interac-etransfer/send-money/${PageIds.DetailsPageId}`);
            }}
          >
            Send another transfer
          </Button>
        </Col>
      </Row>
      <Row>
        <Col md={8}>
          <h4>
            It&apos;s safe and easy to get Interac e-Transfer funds deposited
            directly into your account
          </h4>
          <a href="/" className="text-danger">
            Register for Autodeposit
            {'>'}
          </a>
        </Col>
        <Col md={4}>
          <div style={{ width: '100%', height: 150, border: '1px dotted grey' }}>
            p
          </div>
        </Col>
      </Row>
    </>
  );
};

export default TransferSentPage;
