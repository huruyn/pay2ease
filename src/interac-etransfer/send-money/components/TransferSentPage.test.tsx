import { BrowserRouter } from 'react-router-dom';
import { cleanup, render, screen } from '@testing-library/react';
import TransferSentPage from './TransferSentPage';
import TRANSFER_INFORMATION from '../../../stories/TransferInformation';
import { isValidEmail } from '../../helpers/Validators';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

const component = (
  <BrowserRouter>
    <TransferSentPage
      isCompleted
      transferInformation={TRANSFER_INFORMATION}
    />
  </BrowserRouter>
);

beforeEach(cleanup);

describe('Transfer Sent Page Component', () => {
  it('should render TransferSentPage', () => {
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });

  it('Click Send another transfer button on Transfer Sent Page', () => {
    render(component);
    screen.getAllByRole('button')[1].click();
    expect(mockedUsedNavigate).toBeCalledTimes(1);
  });

  it('Click Check Status Button on Transfer Sent Page', () => {
    render(component);
    screen.getAllByRole('button')[0].click();
    expect(mockedUsedNavigate).toHaveBeenCalled();
  });

  it('should render with the valid values', () => {
    const { container } = render(component);
    expect(container).toMatchSnapshot();

    const sourceEmail = screen.getByText(TRANSFER_INFORMATION.source.email);
    const destinationEmail = screen.getByText(TRANSFER_INFORMATION.destination.email);

    expect(screen.getByText(TRANSFER_INFORMATION.source.name)).toBeInTheDocument();
    expect(screen.getByText(TRANSFER_INFORMATION.destination.name)).toBeInTheDocument();
    expect(screen.getByText(TRANSFER_INFORMATION.fromAccount)).toBeInTheDocument();
    expect(screen.getByText(TRANSFER_INFORMATION.securityQuestion)).toBeInTheDocument();
    expect(screen.getByText(TRANSFER_INFORMATION.referenceNumber.toString())).toBeInTheDocument();
    expect(screen.getByText('$1,000.00')).toBeInTheDocument();
    expect(screen.getByText('$2,000.00')).toBeInTheDocument();
    expect(screen.getByText('Aug 6, 2021')).toBeInTheDocument();
    expect(screen.getByText(/.*Mon, Dec 6, 2021, 1:07 PM.*$/)).toBeInTheDocument();
    expect(isValidEmail(sourceEmail.textContent!));
    expect(isValidEmail(destinationEmail.textContent!));
  });
});
