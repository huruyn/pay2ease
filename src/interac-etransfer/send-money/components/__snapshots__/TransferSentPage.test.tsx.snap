// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transfer Sent Page Component should render TransferSentPage 1`] = `
<div>
  <div
    class="d-flex align-items-center"
  >
    <div
      style="width: 20px; height: 25px; border: 1px dotted grey; text-align: center; margin-right: 10px;"
    >
      P
    </div>
    <h4>
      Your transfer has been sent
    </h4>
  </div>
  <p>
    If you opted to send a notification of your transfer, your contact will receive it shortly.
  </p>
  <div
    class="mt-4 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        From:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        source name
      </label>
      <p>
        source@zippy.cash
      </p>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        To:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        destination name
      </label>
      <p>
        destination@zippy.cash
      </p>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        Account Balance:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        $2,000.00
      </label>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        Security Question:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        Security Question
      </label>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        Response:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        ***************
      </label>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        Amount:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        $1,000.00
      </label>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        From Account:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        392 Jones (8000 001 000000000)
      </label>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        Expiry Date:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        Aug 6, 2021
      </label>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        Reference Number:(keep for your records)
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        123456789
      </label>
    </div>
  </div>
  <p
    class="mt-4"
  >
    Submitted: Mon, Dec 6, 2021, 1:07 PM.
  </p>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-4 mb-4 row"
  >
    <div
      class="d-flex justify-content-end col"
    >
      <button
        class="d-flex btn btn-outline-danger"
        style="margin-right: 10px;"
        type="button"
      >
        Check Status
      </button>
      <button
        class="zippy-btn d-flex btn btn-primary"
        type="button"
      >
        Send another transfer
      </button>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-md-8"
    >
      <h4>
        It's safe and easy to get Interac e-Transfer funds deposited directly into your account
      </h4>
      <a
        class="text-danger"
        href="/"
      >
        Register for Autodeposit
        &gt;
      </a>
    </div>
    <div
      class="col-md-4"
    >
      <div
        style="width: 100%; height: 150px; border: 1px dotted grey;"
      >
        p
      </div>
    </div>
  </div>
</div>
`;

exports[`Transfer Sent Page Component should render with the valid values 1`] = `
<div>
  <div
    class="d-flex align-items-center"
  >
    <div
      style="width: 20px; height: 25px; border: 1px dotted grey; text-align: center; margin-right: 10px;"
    >
      P
    </div>
    <h4>
      Your transfer has been sent
    </h4>
  </div>
  <p>
    If you opted to send a notification of your transfer, your contact will receive it shortly.
  </p>
  <div
    class="mt-4 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        From:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        source name
      </label>
      <p>
        source@zippy.cash
      </p>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        To:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        destination name
      </label>
      <p>
        destination@zippy.cash
      </p>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        Account Balance:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        $2,000.00
      </label>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        Security Question:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        Security Question
      </label>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        Response:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        ***************
      </label>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        Amount:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        $1,000.00
      </label>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        From Account:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        392 Jones (8000 001 000000000)
      </label>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        Expiry Date:
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        Aug 6, 2021
      </label>
    </div>
  </div>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-4"
    >
      <label
        class="form-label"
      >
        Reference Number:(keep for your records)
      </label>
    </div>
    <div
      class="col-md-8"
    >
      <label
        class="form-label"
      >
        123456789
      </label>
    </div>
  </div>
  <p
    class="mt-4"
  >
    Submitted: Mon, Dec 6, 2021, 1:07 PM.
  </p>
  <hr
    style="height: 1px; width: 100%;"
  />
  <div
    class="mt-4 mb-4 row"
  >
    <div
      class="d-flex justify-content-end col"
    >
      <button
        class="d-flex btn btn-outline-danger"
        style="margin-right: 10px;"
        type="button"
      >
        Check Status
      </button>
      <button
        class="zippy-btn d-flex btn btn-primary"
        type="button"
      >
        Send another transfer
      </button>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-md-8"
    >
      <h4>
        It's safe and easy to get Interac e-Transfer funds deposited directly into your account
      </h4>
      <a
        class="text-danger"
        href="/"
      >
        Register for Autodeposit
        &gt;
      </a>
    </div>
    <div
      class="col-md-4"
    >
      <div
        style="width: 100%; height: 150px; border: 1px dotted grey;"
      >
        p
      </div>
    </div>
  </div>
</div>
`;
